server/routes/referrals/completionDeadlineForm.ts

```diff
@@ -0,0 +1,82 @@
+import { Request } from 'express'
+import { body, Result, SanitizationChain, ValidationError, validationResult } from 'express-validator'
+import { DraftReferral } from '../../services/interventionsService'
+import CalendarDay from '../../utils/calendarDay'
+
+export default class CompletionDeadlineForm {
+  private constructor(private readonly request: Request, private readonly result: Result<ValidationError>) {
+    this.result = validationResult(request)
+  }
+
+  static async createForm(request: Request): Promise<CompletionDeadlineForm> {
+    await Promise.all(this.validations.map(validation => validation.run(request)))
+
+    const result = validationResult(request)
+
+    return new CompletionDeadlineForm(request, result)
+  }
+
+  static get validations(): SanitizationChain[] {
+    return [
+      // TODO decide whether we like the fact that express-validator is mutating the request?
+      // e.g. the `bail` before toInt() is so that we don’t end up replaying NaN back to the user
+      // if they enter a blank input
+      body('completion-deadline-day')
+        .notEmpty({ ignore_whitespace: true })
+        .withMessage('The date by which the service needs to be completed must include a day')
```

Comment:
this is good, but i wonder if as we add more and more validation messages it would be useful to contain them in a single file, indexed by description `{ "dateFieldEmtpy": "The date by which the service needs to be completed must include a day" }`. this way we could easily have designers audit the messages to ensure they are user friendly enough.

Reply:
Good shout - added `server/utils/errorMessages.ts`.